extension/src/popup/metrics/views.ts

```diff
@@ -0,0 +1,30 @@
+import { METRIC_NAMES } from "popup/constants/metricsNames";
+
+import { registerHandler, emitMetric } from "helpers/metrics";
+
+import { navigate } from "popup/ducks/views";
+import { AppState } from "popup/App";
+import { ROUTES } from "popup/constants/routes";
+
+const routeToEventName = {
+  [ROUTES.home]: METRIC_NAMES.viewHome,
+  [ROUTES.account]: METRIC_NAMES.viewAccount,
+  [ROUTES.signTransaction]: METRIC_NAMES.viewSignTransaction,
+  [ROUTES.grantAccess]: METRIC_NAMES.viewGrantAccess,
+  [ROUTES.mnemonicPhrase]: METRIC_NAMES.viewMnemonicPhrase,
+  [ROUTES.unlockAccount]: METRIC_NAMES.viewUnlockAccount,
+  [ROUTES.mnemonicPhraseConfirmed]: METRIC_NAMES.viewMnemonicPhraseConfirmed,
+  [ROUTES.createPassword]: METRIC_NAMES.viewCreatePassword,
+  [ROUTES.recoverAccount]: METRIC_NAMES.viewRecoverAccount,
+};
+
+registerHandler<AppState>(navigate, (_, a) => {
+  // Awkward, but gives us types on action payload
+  const action = a as ReturnType<typeof navigate>;
```

Comment:
`as` here just sorta sidesteps the type guarantees and says "yea i'm pretty sure i know what this is", but I couldn't quickly figure out a way to get the types to pipe through correctly, and I wanted to be able to rely on autocomplete for the action payload